driver/mloMarginRakningLossHost.hpp

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/tensor.hpp>
+#include <miopen/tensor_view_utils.hpp>
+#include <../test/ford.hpp>
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloMarginRankingLossForwardRunHost(const miopenTensorDescriptor_t input1Desc,
+                                           const Tgpu* input1,
+                                           const miopenTensorDescriptor_t input2Desc,
+                                           const Tgpu* input2,
+                                           const miopenTensorDescriptor_t targetDesc,
+                                           const Tgpu* target,
+                                           const miopenTensorDescriptor_t outputDesc,
+                                           Tcheck* output,
+                                           float margin,
+                                           float divisor,
```

Comment:
`size_t`

Reply:
CL kernel use float, and I find no reason to change it

Reply:
There are cases the number of elements inside input tensor are too big for fully presented by fp32. For example, `nelem = 214748364`, its value cannot be presented fully with `float`, that why it you should use `integer` variable. Beside, the number of elements can be so large, more than max value of `uint_32t`, therefore, `size_t` and `uint64_t` are the best choice for this variable.

Let's take a look at [this script](https://onlinegdb.com/fllzqab8w)
<img width="706" alt="image" src="https://github.com/user-attachments/assets/3bee7abf-9a58-4eaf-93a4-d88389f36b5e">

Reply:
1.0 you use maybe still int. The problem is that the int y will also be convert to float_accum to be divided by loss_sum. If you converted divisor before (by CPU) you can make a little improvement than do it in GPU. That may be the reason for CL code still using float
<img width="540" alt="image" src="https://github.com/user-attachments/assets/c8fe08d6-5a9a-4712-9506-d939d438955c">

Reply:
resolved